audit/audit.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package audit provides an audit log writer for access to secrets.
+package audit
+
+import (
+	"encoding/json"
+	"io"
+	"net/netip"
+	"os"
+
+	"github.com/tailscale/setec/acl"
+	"github.com/tailscale/setec/types/api"
+)
+
+// Principal is the identity of a client taking action on the secrets
+// service.
+type Principal struct {
+	Hostname string     `json:"hostname"`
+	IP       netip.Addr `json:"ip"`
+	User     string     `json:"user,omitempty"`
+	Tags     []string   `json:"tags,omitempty"`
+}
+
+// Entry is an audit log entry.
+type Entry struct {
+	// Principal is the client who is doing something.
```

Comment:
Should we include a version marker in the wire format for this?

Reply:
My general approach here is to not have a version marker for v1, and if we ever have a v2, it can have a version field whose existence indicates v2-ness. Does that work?

Reply:
Yup, that seems fine. As you noted, with JSON we don't have to be extremely sensitive about presence or absence.